phasehq/kubernetes-secrets-operator
# Feat: add service account support

How to test things locally:
- Start a new kubernetes cluster - `minikube start`
- Apply the CRD - kubectl apply -f kubectl apply -f crd-template.yaml
- Create the Phase service token v2 secret
```fish
kubectl create secret generic phase-service-token \
  --from-literal=token=<TOKEN> \
  --type=Opaque \
  --namespace=default
```
- Apply the following CR:
```yaml
apiVersion: secrets.phase.dev/v1alpha1
kind: PhaseSecret
metadata:
  name: example-phase-secret
  namespace: default
spec:
  phaseApp: "example-app" # The name of your Phase application
  phaseAppEnv: "development" # OPTIONAL The Phase application environment to fetch secrets from
  phaseAppEnvPath: "/" # OPTIONAL Path within the Phase application environment to fetch secrets from
  phaseHost: "http://localhost" # OPTIONAL - URL of a Phase Console instance
  authentication:
    serviceToken:
      serviceTokenSecretReference:
        secretName: "phase-service-token" # Name of the Phase service token with access to your application
        secretNamespace: "default"
  managedSecretReferences: # Managed secrets in Kubernetes that Phase will sync secrets with
    - secretName: "my-application-secret" # Name of the managed secret in Kubernetes that will be consumed by your application
      secretNamespace: "default"
```
- In python venv - install all dependencies
- Run the operator in dev mode - `kopf run src/main.py`
- Watch for secrets being created - `watch kubectl get secrets `
- Verify that the secret has been synced - `kubectl get secret my-application-secret -o yaml`
